python/nano/test/openvino/pytorch/test_openvino.py

```diff
@@ -81,3 +82,27 @@ def test_openvino_inputsample_from_trainloader(self):
         trainer.fit(pl_model, dataloader)
         assert pl_model.forward == pl_model._torch_forward 
         assert  pl_model.ov_infer_engine is None
+
+    def test_trainer_trace_openvino(self):
+        trainer = Trainer(max_epochs=1)
+        model = mobilenet_v3_small(num_classes=10)
+        
+        x = torch.rand((10, 3, 256, 256))
+        y = torch.ones((10, ), dtype=torch.long)
+
+        # trace a torch model 
+        openvino_model = trainer.trace(model, x, 'openvino')
+        y_hat = openvino_model(x)
+        assert y_hat.shape == (10, 10)
+
+        # trace pytorch-lightning model
+        pl_model = Trainer.compile(model, loss=torch.nn.CrossEntropyLoss(),
+                                   optimizer=torch.optim.SGD(model.parameters(), lr=0.01),
+                                   openvino=True)
```

Comment:
remove the `openvino` parameter from `Trainer.compile`, and just call `bind_openvino_methods` unconditionally

Reply:
This ut should be removed after we make sure all our new APIs work.  
I didn't remove the old API using compile to to openvino optimization.  
Shall we remove it in this PR?

Reply:
We should switch to new OpenVINO APIs in this PR.